src/Form/DataTransformer/ModelsToArrayTransformer.php

```diff
@@ -125,6 +130,11 @@ public function __unset($name)
         unset($this->$name);
     }
 
+    /**
+     * @return string[]
+     *
+     * @phpstan-param T[] $collection
+     */
     public function transform($collection)
     {
         if (null === $collection) {
```

Comment:
I'm not sure if we should check for invalid types in the data transformer classes.

Should we *trust* the typing system or should we check for every technical possible values.

Reply:
Same for the other transfomers

Reply:
As explained here: https://github.com/sonata-project/SonataAdminBundle/pull/6361#discussion_r483943766

The Phpdoc from Symfony is saying that we must deal empty value, which is generally `null`.

What I understand is that empty values are valid values.
So it's not about trust the typing system, it's about fixing in.

Phpdoc should be `@phpstan-param T[]|null $collection` because the `null` case can occur.

Reply:
I like the way it's done here: 
https://github.com/symfony/form/blob/master/Extension/Core/DataTransformer/StringToFloatTransformer.php
Or 
https://github.com/symfony/form/blob/master/Extension/Core/DataTransformer/ChoicesToValuesTransformer.php

Reply:
I changed this to always except null values for both transformations.